src/sysctl/sysctl.c

```diff
@@ -106,16 +106,16 @@ static int parse_file(OrderedHashmap *sysctl_options, const char *path, bool ign
 
         log_debug("Parsing %s", path);
         for (;;) {
-                char l[LINE_MAX], *p, *value, *new_value, *property, *existing;
+                char *p, *value, *new_value, *property, *existing;
+                _cleanup_free_ char *l = NULL;
                 void *v;
-                int k;
-
-                if (!fgets(l, sizeof(l), f)) {
-                        if (feof(f))
-                                break;
+                int k, ret;
+                ret = read_line(f, LONG_LINE_MAX, &l);
+                if (ret == 0)
+                        break;
 
+                if (ret < 0)
                         return log_error_errno(errno, "Failed to read file '%s', ignoring: %m", path);
```

Comment:
pass "ret" (or 'k' …) as first argument to log_error_errno() now, please, as that's how read_line() returns its errors (all our own APIs return error codes directly, rather than use errno.